File: Lab_2/task_3/3.py
# Функция для переворачивания числа и удаления ведущих нулей
def reverse_number(num):
    reversed_num = 0
    while num > 0:
        reversed_num = reversed_num * 10 + num % 10
        num //= 10
    return reversed_num

def main():
    n = int(input("Введите количество чисел: "))

    if n <= 0:
        print("Некорректный ввод.")
        return

    numbers = []
    for i in range(n):
        num = int(input(f"Введите число {i + 1}: "))
        numbers.append(reverse_number(num))

    print("Результаты:", " ".join(map(str, numbers)))

if __name__ == "__main__":
    main()

File: Lab_2/task_3/3.js
// Функция для переворачивания числа и удаления ведущих нулей
function reverseNumber(num) {
    let reversed = 0;
    while (num > 0) {
        reversed = reversed * 10 + num % 10;
        num = Math.floor(num / 10);
    }
    return reversed;
}

function main() {
    const n = parseInt(prompt("Введите количество чисел: "));

    if (isNaN(n) || n <= 0) {
        console.log("Некорректный ввод.");
        return;
    }

    let numbers = [];
    for (let i = 0; i < n; i++) {
        const num = parseInt(prompt(`Введите число ${i + 1}: `));
        if (!isNaN(num)) {
            numbers.push(reverseNumber(num));
        }
    }

    console.log("Результаты: ", numbers.join(" "));
}

main();
